html/browsers/history/the-location-interface/location-ancestororigins-involved.html

```diff
@@ -0,0 +1,169 @@
+<!doctype html>
+<script src=/resources/testharness.js></script>
+<script src=/resources/testharnessreport.js></script>
+<script src=/common/get-host-info.sub.js></script>
+<div id=log></div>
+<script>
+const embedPath = new URL("resources/ancestororigins-embed.py", location.href).pathname,
+      info = get_host_info(),
+      localOrigin = info.HTTP_ORIGIN,
+      localEmbed = localOrigin + embedPath,
+      remoteOrigin = info.HTTP_REMOTE_ORIGIN,
+      remoteEmbed = remoteOrigin + embedPath,
+      remoteOrigin2 = info.HTTP_REMOTE_ORIGIN_WITH_DIFFERENT_PORT
+      remoteEmbed2 = remoteOrigin2 + embedPath
+let id = 0
+
+async_test(t => {
+  const iframe = document.createElement("iframe"),
+        localId = ++id
+  iframe.sandbox = "allow-scripts"
+  iframe.src = localEmbed + "?iframe=|" + remoteEmbed + "?id=" + localId
+  document.body.appendChild(iframe)
+  t.add_cleanup(() => iframe.remove())
+
+
+  self.addEventListener("message", t.step_func(e => {
+    if(e.data.id === localId) {
+      assert_array_equals(e.data.output, ["null", self.location.origin])
+      t.done()
+    }
+  }))
+}, "Ensure sandboxed iframes show up as null")
+
+;[
+  {
+    outerPolicy: true,
+    desc: "A uses no-referrer -> B -> A",
+    results: [remoteOrigin, "null"]
+  },
+  {
+    outerPolicy: true,
+    innerEmbed: remoteEmbed2,
```

Comment:
This array could use some docs describing what the various fields are.

Also, the `results` vs `innerResults` naming is weird, because `innerResults` is outer compared to `results`.  Maybe `intermediateResults` instead of `innerResults`?